src/main/kotlin/net/pdevita/creeperheal2/CreeperHeal2.kt

```diff
@@ -68,7 +71,16 @@ class CreeperHeal2 : JavaPlugin() {
             return null
         }
 
-        val newBlockList = LinkedList(blockList.filter { settings.blockList.allowMaterial(it.type) })
+        val newBlockList = LinkedList<Block>()
+
+        for (block in blockList) {
+            if (settings.blockList.allowMaterial(block.type)) {
+                newBlockList.add(block)
+            } else {
+                block.breakNaturally()
+                block.world.dropItemNaturally(block.location, ItemStack(block.type))
+            }
```

Comment:
As far as I understand, as long as CreeperHeal ignores the block and doesn't touch it, the explosion should handle this for us, and that would be preferable as [explosions don't always drop blocks as items](https://minecraft.fandom.com/wiki/Explosion#Dropping_blocks)). The main blacklist filtering occurs in the explosion class during its setup. Was it not working? What kind of behavior were you seeing?